src/main/java/dev/andrylat/vomelianchuk/card_number_operations/CardNumberValidator.java

```diff
@@ -0,0 +1,51 @@
+package dev.andrylat.vomelianchuk.card_number_operations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CardNumberValidator {
+    private static final String HIATUS_SEPARATOR = "\\s";
+    private static final int LENGTH_CARD_NUMBER = 16;
+    private static final String ERROR_NULL_CARD_NUMBER = "Error receiving null object instead of card number";
+    private static final String ERROR_LENGTH_CARD = "Length should be 16 symbols";
+    private static final String ERROR_ONLY_DIGITS = "Number should contain only digits";
+
+    private final List<String> errors = new ArrayList<>();
+
+    public void validate(String cardNumber) throws CardValidationException {
+        if (cardNumber != null) {
+            cardNumber = transform(cardNumber);
+            checkLengthCardNumber(cardNumber);
+            checkSymbolsCardNumber(cardNumber);
+            new LuhnAlgorithm().checkLastNumber(cardNumber, errors);
+        } else {
+            errors.add(ERROR_NULL_CARD_NUMBER);
+            throw new CardValidationException(errors);
+        }
+    }
+
+    private String transform(String cardNumber) {
```

Comment:
minor: transform is very generaic name. Better to use something like "removeSeparators"